geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/collection/CloseableIterator.scala

```diff
@@ -62,12 +62,23 @@ trait CloseableIterator[+A] extends Iterator[A] with Closeable {
 
   override def filter(p: A => Boolean): CloseableIterator[A] = CloseableIterator(super.filter(p), self.close())
 
+  override def ++[B >: A](that: => GenTraversableOnce[B]): CloseableIterator[B] = {
```

Comment:
Ok, hate to do it... can you add a quick unit test to CloseableIteratorTest?  

We could use an intern.